src/Client/MainComponents/Widgets.fs

```diff
@@ -215,10 +215,7 @@ type Widget =
         let isProtocolSearch, setProtocolSearch = React.useState(true)
         let filteredTemplates = Protocol.Search.filterTemplates (templates, config)
         if model.ProtocolState.TemplatesSelected.Length > 0 && (fst importTypeStateData).SelectedColumns.Length = 0 then
-            let columns =
-                model.ProtocolState.TemplatesSelected
-                |> List.map (fun template -> Array.init template.Table.Columns.Length (fun _ -> true))
-                |> Array.ofList
+            let columns = Array.init model.ProtocolState.TemplatesSelected.Length (fun _ -> Set.empty<int>)
```

Comment:
You could also do double index. So:

```fsharp
let columns = Set.empty<int,int>
```

where first int is template index and second int is column index. This would avoid initiating anything that is not used.

In addition `columns` might now be `skippedColumns` if i understand this correctly